### PR TITLE
ESQL: Use specific version in security IT

### PR DESCRIPTION
**Pull request by alex-spies**
Do not use the SNAPSHOT language version in RemoteClusterSecurityEsqlIT.

Initially, I wanted to keep this on the latest language version so that its tests automatically fail when we make breaking changes to the language; since testing language features is not the point of these tests, I think it's better to keep the version on `2024.04.01` instead.

### CODE REVIEW

File: x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java [hunk 1]
```diff
@@ -701,7 +701,8 @@ public static Map<String, Object> runEsqlSync(
         RequestOptions.Builder options = request.getOptions().toBuilder();
         options.setWarningsHandler(WarningsHandler.PERMISSIVE); // We assert the warnings ourselves
         options.addHeader("Content-Type", mediaType);
-        if ("true".equals(System.getProperty("tests.version_parameter_unsupported"))) {
+
+        if (EsqlSpecTestCase.availableVersions().isEmpty()) {
```
Comment (alex-spies) [inline x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java:705]: Unrelated cleanup

**Review by nik9000 — APPROVED** [review]
